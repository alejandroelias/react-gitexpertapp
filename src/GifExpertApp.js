import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GigGrid } from './components/GigGrid';

export const GifExpertApp = ({defaultCategories = []}) => {


    const [categories, setCategories] = useState(defaultCategories);
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories = { setCategories }/>
            <hr />

            <ol>
            {
                categories.map( category => (
                    <GigGrid
                        key = { category }
                        category = { category }
                    />
                ))
            }
            </ol>
        </>
    )
}
