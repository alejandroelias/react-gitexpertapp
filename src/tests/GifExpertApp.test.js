import { shallow } from 'enzyme'
import React from 'react'
import { GifExpertApp } from '../GifExpertApp'

describe('Test in <GifExpertApp />', () => {
    
    test('should show the correct component and math with the snapshot', () => {
        
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show list of categories', () => {
        
        const categories = ['Rick and Morty', 'One Punch'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GigGrid').length).toBe(categories.length+1);
    })
    
    
})
