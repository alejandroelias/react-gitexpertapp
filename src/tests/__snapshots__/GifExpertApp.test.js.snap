// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifExpertApp /> should show list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GigGrid
      category="Rick and Morty"
      key="Rick and Morty"
    />
    <GigGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;

exports[`Test in <GifExpertApp /> should show the correct component and math with the snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
